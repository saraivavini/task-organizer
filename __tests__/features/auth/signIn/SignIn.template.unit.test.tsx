import { SignInTemplate } from '../../../../src/features/auth/signIn/SignIn.template';

import { componentBuilder } from '../../../utils/componentBuilder';

const { renderComponent } = componentBuilder(SignInTemplate, {
  username: {
    value: '',
    onChange: jest.fn(),
  },
  password: {
    value: '',
    onChange: jest.fn(),
  },
  onSubmit: jest.fn(),
});

describe('SignIn template tests', () => {
  it('Should renders correctly', () => {
    const { component } = renderComponent();

    expect(component.toJSON()).toMatchSnapshot();
  });
});
