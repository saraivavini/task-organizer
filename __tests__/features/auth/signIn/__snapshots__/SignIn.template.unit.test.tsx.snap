// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn template tests Should renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 24,
      }
    }
  >
    <Image
      accessibilityLabel="logo-image"
      alt="logo-image"
      dataSet={Object {}}
      onError={[Function]}
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 96,
          "maxWidth": "100%",
          "width": 96,
        }
      }
    />
    <Text
      dataSet={Object {}}
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": undefined,
          "fontSize": 24,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 36,
          "marginTop": 64,
          "textDecorationLine": undefined,
        }
      }
    >
      auth.signIn.title
    </Text>
    <View
      dataSet={Object {}}
      style={
        Object {
          "marginTop": 64,
        }
      }
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "width": "100%",
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <View
            dataSet={Object {}}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderBottomLeftRadius": 8,
                  "borderColor": "#d4d4d4",
                  "borderRightWidth": 0,
                  "borderTopLeftRadius": 8,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                  "width": "20%",
                },
                Array [
                  Object {},
                ],
              ]
            }
          >
            <Text />
          </View>
          <View
            dataSet={Object {}}
            style={Object {}}
          />
          <View
            dataSet={Object {}}
            isFocused={false}
            style={
              Object {
                "alignItems": "center",
                "borderBottomLeftRadius": 0,
                "borderBottomRightRadius": 8,
                "borderColor": "#d4d4d4",
                "borderRadius": 4,
                "borderTopLeftRadius": 0,
                "borderTopRightRadius": 8,
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "marginLeft": 0,
                "overflow": "hidden",
              }
            }
          >
            <TextInput
              accessible={true}
              dataSet={Object {}}
              editable={true}
              onBlur={[Function]}
              onChangeText={[MockFunction]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Username"
              placeholderTextColor="#a3a3a3"
              secureTextEntry={false}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderBottomLeftRadius": 0,
                  "borderTopLeftRadius": 0,
                  "color": "#171717",
                  "flex": 1,
                  "fontFamily": undefined,
                  "fontSize": 20,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "height": "100%",
                  "paddingBottom": 8,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 8,
                  "width": "100%",
                }
              }
              value=""
            />
          </View>
        </View>
      </View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "height": 24,
          }
        }
      />
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "width": "100%",
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <View
            dataSet={Object {}}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderBottomLeftRadius": 8,
                  "borderColor": "#d4d4d4",
                  "borderRightWidth": 0,
                  "borderTopLeftRadius": 8,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                  "width": "20%",
                },
                Array [
                  Object {},
                ],
              ]
            }
          >
            <Text />
          </View>
          <View
            dataSet={Object {}}
            style={Object {}}
          />
          <View
            dataSet={Object {}}
            isFocused={false}
            style={
              Object {
                "alignItems": "center",
                "borderBottomLeftRadius": 0,
                "borderBottomRightRadius": 8,
                "borderColor": "#d4d4d4",
                "borderRadius": 4,
                "borderTopLeftRadius": 0,
                "borderTopRightRadius": 8,
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "marginLeft": 0,
                "overflow": "hidden",
              }
            }
          >
            <TextInput
              accessible={true}
              dataSet={Object {}}
              editable={true}
              onBlur={[Function]}
              onChangeText={[MockFunction]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Password"
              placeholderTextColor="#a3a3a3"
              secureTextEntry={true}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderBottomLeftRadius": 0,
                  "borderTopLeftRadius": 0,
                  "color": "#171717",
                  "flex": 1,
                  "fontFamily": undefined,
                  "fontSize": 20,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "height": "100%",
                  "paddingBottom": 8,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 8,
                  "width": "100%",
                }
              }
              value=""
            />
          </View>
        </View>
      </View>
    </View>
    <View
      dataSet={Object {}}
      style={
        Object {
          "marginTop": 64,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        dataSet={Object {}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#006AFF",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
          test={true}
        >
          <View
            dataSet={Object {}}
            style={Object {}}
          >
            <Text
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#FFFFFF",
                  "fontFamily": undefined,
                  "fontSize": 18,
                  "fontStyle": "normal",
                  "fontWeight": "500",
                  "letterSpacing": 0,
                  "lineHeight": 27,
                  "textDecorationLine": undefined,
                }
              }
            >
              auth.signIn.buttonLabel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
